Sazabi.cpp

```diff
@@ -3636,6 +3666,39 @@ CString CSazabi::GetChromiumVersionStr()
 	return strRet;
 }
 
+std::tuple<WORD, WORD, WORD, WORD> CSazabi::GetFileVersion(CString filePath)
+{
+	if (!::PathFileExists(filePath))
+	{
+		return std::make_tuple(0, 0, 0, 0);
+	}
+
+	DWORD handle = 0;
+	DWORD versionSize = ::GetFileVersionInfoSize((LPCWSTR)filePath, &handle);
+
+	if (versionSize == 0)
+	{
+		return std::make_tuple(0, 0, 0, 0);
+	}
+
+	PBYTE pData = new BYTE[versionSize];
+	memset(pData, 0x00, versionSize);
+	if (::GetFileVersionInfo((LPCWSTR)filePath, NULL, versionSize, pData))
+	{
+		VS_FIXEDFILEINFO* fileInfo = nullptr;
+		UINT len = 0;
+		if (VerQueryValue(pData, _T("\\"), (PVOID*)&fileInfo, &len))
+		{
+			return std::make_tuple(HIWORD(fileInfo->dwFileVersionMS),

```

Comment:
pData leaks here?

Reply:
Ah... I will fix it.

Reply:
Fixed.

Reply:
https://learn.microsoft.com/en-us/windows/win32/api/winver/nf-winver-verqueryvaluew

> When this method returns, contains the address of a pointer to the requested version information in the buffer pointed to by pBlock. The memory pointed to by lplpBuffer is freed when the associated pBlock memory is freed.

Does it safe?

Reply:
It might be more safer like this:

```c++
WORD wHiMS = HIWORD(fileInfo->dwFileVersionMS);
WORD wLoMS = LOWORD(fileInfo->dwFileVersionMS);
WORD wHiLS = HIWORD(fileInfo->dwFileVersionLS);
WORD wLoLS = LOWORD(fileInfo->dwFileVersionLS);
delete[] pData;
return std::make_tuple(HIWORD(fileInfo->dwFileVersionMS),....
```

Reply:
As you say, it seems not be safe. Thanks.

I fixed to delete `pData` after create tuple.